src/ucs/datastruct/bitmap.h

```diff
@@ -661,6 +669,181 @@ ucs_bitmap_bits_is_zero(const ucs_bitmap_word_t *bits, size_t num_words)
     return 1;
 }
 
+
+/* Helper function to count the number of bits that are set to 1 */
+static UCS_F_ALWAYS_INLINE size_t
+ucs_bitmap_bits_popcount(const ucs_bitmap_word_t *bits, size_t num_words)
+{
+    size_t popcount = 0;
+    const ucs_bitmap_word_t *bits_word;
+
+    ucs_carray_for_each(bits_word, bits, num_words) {
+        popcount += ucs_popcount(*bits_word);
+    }
+
+    return popcount;
+}
+
+
+/* Helper function to return the number of bits that are set to 1 upto a
+   given bit index */
+static UCS_F_ALWAYS_INLINE size_t ucs_bitmap_bits_popcount_upto_index(
+        const ucs_bitmap_word_t *bits, size_t num_words, size_t bit_index)
+{
+    size_t last_word       = bit_index / UCS_BITMAP_BITS_IN_WORD;
+    ucs_bitmap_word_t mask = ucs_bitmap_word_bit_mask(bit_index) - 1;
+
+    UCS_BITMAP_CHECK_INDEX(bit_index, num_words, <=);
+    return ucs_bitmap_bits_popcount(bits, last_word) +
+           ((mask == 0) ? 0 : ucs_popcount(bits[last_word] & mask));
+}
+
+
+/* Helper function to copy between bitmap bit arrays. Destination size must
+   be at lease the source size */
+static UCS_F_ALWAYS_INLINE void
+ucs_bitmap_bits_copy(const ucs_bitmap_word_t *src_bits, size_t src_num_words,
```

Comment:
I noticed that in other functions destination params come first, followed by source ones.
In this one the order is the opposite.
Maybe it makes sense to maintain the same order `(dst, src)` in all similar functions to avoid confusion?

Reply:
right. but actually this function can be moved to a later PR since it's needed only by static_bitmap